parlai/core/metrics.py

```diff
@@ -411,26 +444,43 @@ def normalize_answer(s):
     return s
 
 
-def aggregate_named_reports(named_reports: Dict[str, Dict[str, Metric]]):
+def aggregate_named_reports(
+    named_reports: Dict[str, Dict[str, Metric]], micro_average: bool = False
+):
     """
     Aggregate metrics from multiple reports.
 
     :param reports: Dict of tasks -> metrics.
+    :param micro_average: If true, top level metrics will be the macro average.
     """
+    if len(named_reports) == 0:
+        raise ValueError("Cannot aggregate empty reports.")
+    if len(named_reports) == 1:
+        # no real aggregation to be done
+        return next(iter(named_reports.values()))
+
     # reporters is a list of teachers or worlds
     m: Dict[str, Metric] = {}
+    macro_averages: Dict[str, List[Metric]] = {}
     for task_id, task_report in named_reports.items():
         for each_metric, value in task_report.items():
             if value.is_global():
                 # just take the first one we saw
                 if each_metric not in m:
                     m[each_metric] = value
             else:
-                # none + a => a from implementation of Metric.__add__
-                m[each_metric] = m.get(each_metric) + value
-                if len(named_reports) > 1:
-                    task_metric = f'{task_id}/{each_metric}'
-                    m[task_metric] = m.get(task_metric) + value
+                task_metric = f'{task_id}/{each_metric}'
+                m[task_metric] = m.get(task_metric) + value
+                if micro_average or isinstance(value, (SumMetric, FixedMetric)):
```

Comment:
Ah, here's the `SumMetric` carve-out. Since we're implementing macro-*averaging* here, why don't we have the condition be that you only macroaverage if it's an `AverageMetric`?

Reply:
Yeah I wrestled with that. PPL stands out as a weird edge case. It is an average metric, so it would work. What does @klshuster think?

Reply:
Updated per your suggestion.